src/EventFactory.php

```diff
@@ -50,30 +50,26 @@ final class EventFactory implements EventFactoryInterface
     public function __construct(SerializerInterface $serializer, RepresentationSerializerInterface $representationSerializer, Options $options, string $sdkIdentifier, string $sdkVersion)
     {
         $this->serializer = $serializer;
-        $this->representationSerializer = $representationSerializer;
         $this->options = $options;
         $this->sdkIdentifier = $sdkIdentifier;
         $this->sdkVersion = $sdkVersion;
+        $this->stacktraceBuilder = new StacktraceBuilder($options, $representationSerializer);
```

Comment:
This is still a hidden dependency like this... How can we improve this?

Reply:
One of the things I learned in `2.x` is that the way an event data is collected or constructed is often something internal and the user shouldn't worry about, so I don't think we should expose such dependencies. Also, making the builders part of the public API will probably get us back to the point where it will be harder to improve the API or do bug fixes because we can't break BC